pyrealm/phenology/fapar_limitation.py

```diff
@@ -0,0 +1,305 @@
+        annual_total_precip: NDArray[np.float64],
+        aridity_index: NDArray[np.float64],
+    ) -> None:
+        r"""Annual peak fractional absorbed photosynthetically active radiation (fAPAR).
+
+        Computes fAPAR_max as minimum of a water-limited and an energy-limited
+        quantity.
+
+        Args:
+            annual_total_potential_gpp: Aka A_0, the annual sum of potential GPP.
+              Potential GPP would be achieved if fAPAR = 1. [mol m^{-2} year^{-1}]
+            annual_mean_ca: Ambient CO2 partial pressure. [Pa]
+            annual_mean_chi: Annual mean ratio of leaf-internal CO2 partial pressure to
+               c_a during the growing season (>0℃). [Pa]
+            annual_mean_vpd: Aka D, annual mean vapour pressure deficit (VPD) during the
+              growing season (>0℃). [Pa]
+            annual_total_precip: Aka P, annual total precipitation. [mol m^{-2}
+             year^{-1}]
+            aridity_index: Aka AI, climatological estimate of local aridity index.
+        """
+
+        self.annual_total_potential_gpp = annual_total_potential_gpp
+        """Aka A_0, the annual sum of potential GPP. [mol m^{-2} year^{-1}]"""
+        self.annual_mean_ca = annual_mean_ca
+        """Ambient CO2 partial pressure. [Pa]"""
+        self.annual_mean_chi = annual_mean_chi
+        """Annual mean ratio of leaf-internal CO2 partial pressure to c_a during the 
+        growing season (>0℃). [Pa]"""
+        self.annual_mean_vpd = annual_mean_vpd
+        """Annual mean vapour pressure deficit (VPD) during the growing season [Pa]"""
+        self.annual_total_precip = annual_total_precip
+        """Annual total precipitation. [mol m^{-2} year^{-1}]"""
+        self.aridity_index = aridity_index
+        """Climatological estimate of local aridity index."""
+
+        self._check_shapes()
+
+        # Constants used for phenology computations
+        self.phenology_const = PhenologyConst()
+
+        #  f_0 is the ratio of annual total transpiration of annual total
+        #  precipitation, which is an empirical function of the climatic Aridity Index
+        #  (AI).
+        a, b, c = self.phenology_const.f0_coefficients
+        f_0 = a * np.exp(-b * np.log(aridity_index / c) ** 2)
+
+        fapar_energylim = 1.0 - self.phenology_const.z / (
+            self.phenology_const.k * annual_total_potential_gpp
+        )
+        fapar_waterlim = (
+            f_0
+            * annual_total_precip
+            * annual_mean_ca
+            * (1 - annual_mean_chi)
+            / (1.6 * annual_mean_vpd * annual_total_potential_gpp)
+        )
+
+        self.fapar_max = np.minimum(fapar_waterlim, fapar_energylim)
+        """Maximum fapar given water or energy limitation."""
+        self.energy_limited = fapar_energylim < fapar_waterlim
+        """Is fapar_max limited by water or energy."""
+        self.annual_precip_molar = annual_total_precip
+        """The annual precipitation in moles."""
```

Comment:
```suggestion
        """Maximum fapar given water or energy limitation for each year."""
        self.energy_limited = fapar_energylim < fapar_waterlim
        """Is fapar_max limited by water or energy for each year."""
        self.annual_precip_molar = annual_total_precip
        """The annual precipitation in moles for each year."""
```
I think it isn't clear that these are annual results and not overall totals.